Server/BackEndMonitoringServer/src/CLogicalDiskStatus.cpp

```diff
@@ -0,0 +1,83 @@
+#include "stdafx.h"
+
+#include "CLogicalDiskStatus.h"
+
+CLogicalDiskStatus* CLogicalDiskStatus::FactoryLogicalDiskStatus(
+	const std::string& disk_name,
+	EMemoryCountType count_type)
+{
+	CLogicalDiskStatus* p_created_object = new CLogicalDiskStatus(
+		disk_name, count_type);
+
+	if (!p_created_object->TryUpdateCurrentStatus())
+	{
+		return nullptr;
+	}
+
+	return p_created_object;
+}
+
+bool CLogicalDiskStatus::TryUpdateCurrentStatus()
+{
+	//try-catch block for avoidind floppy disks
+	try
+	{
+		m_disk_info = std::filesystem::space(m_disk_name);
+	}
+	catch (std::filesystem::filesystem_error&)
+	{
+		return false;
+	}
+	if (m_disk_info.capacity == NULL &&
+		m_disk_info.available == NULL &&
+		m_disk_info.free == NULL)
+	{
+		return false;
+	}
+
+	return true;
+}
+
+std::string CLogicalDiskStatus::GetDiskName() const
```

Comment:
return const string &

Reply:
I need to use this function in other const methode. For this, I need to make this one const. That's why I put const and delete &. Name of disk will be 3 char always, so this's not big problem of return object, I suppose